app/controllers/admin/report_generator_controller.rb

```diff
@@ -1,47 +1,139 @@
 class Admin::ReportGeneratorController < AdminAreaController
   layout 'admin_area'
+  require 'date'
 
   def index
+    if session[:selected_dates] == nil
+      @start_date = 1.month.ago
+      @end_date = DateTime.current
+    else
+      @start_date = DateTime.parse(selected_dates[0])
+      @end_date = DateTime.parse(selected_dates[1])
+    end
   end
 
-# New users/month
+  def select_date_range
+    session[:selected_dates] = nil
+    @start = Date::civil(params[:start][:year].to_i, params[:start][:month].to_i,
+                      params[:start][:day].to_i)
+    @start_date = Time.zone.at(@start.to_time).to_datetime
+
+    @end = Date::civil(params[:end][:year].to_i, params[:end][:month].to_i,
+                      params[:end][:day].to_i)
+    @end_date = Time.zone.at(@end.to_time).to_datetime
+
+    selected_dates << @start_date << @end_date
+    redirect_to :back
+  end
+
+# New users
   def new_users
+    if session[:selected_dates] == nil

```

Comment:
One pattern that rubyist often rely on is that `nil == false`. That way can write `unless session[:selected_dates]` (because `unless` is the same as `if !`